packages/browser-wallet/src/background/confirmation.ts

```diff
@@ -62,57 +101,68 @@ async function monitorCredentialStatus(
 async function startMonitoringCredentialStatus(network: NetworkConfiguration) {
     const creds = await storedCredentials.get(network.genesisHash);
     if (creds) {
-        creds
-            .filter(isPendingCred)
-            .forEach((cred) => monitorCredentialStatus(network.jsonRpcUrl, cred, network.genesisHash));
+        await Promise.all([
+            creds
+                .filter(isPendingCred)
+                .map((cred) => monitorCredentialStatus(network.jsonRpcUrl, cred, network.genesisHash)),
+        ]);
     }
 }
 
+let monitoredIdentities: PendingIdentity[] = [];
+
+const isIdEqualTo = (id1: BaseIdentity) => (id2: BaseIdentity) =>
+    id1.index === id2.index && id1.providerIndex === id2.providerIndex;
+
+const isMonitoringIdentity = (id: PendingIdentity): boolean => monitoredIdentities.find(isIdEqualTo(id)) !== undefined;
```

Comment:
```suggestion
const isMonitoringIdentity = (id: PendingIdentity): boolean => monitoredIdentities.some(identityMatch(id));
```
We already have that helper somewhere called identityMatch.